static/src/javascripts/lib/page.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import $ from 'lib/$';
+import config from 'lib/config';
+import { getBreakpoint } from 'lib/detect';
+
+type yesable = ?(arg: boolean | string) => boolean;
+type noable = ?() => boolean;
+
+const isit = (
+    isTrue: ?(string | boolean),
+    yes: yesable,
+    no: noable,
+    arg: any
+): boolean => {
+    if (isTrue) {
+        return yes ? yes(arg || isTrue) : arg || !!isTrue;
+    }
+
+    return no ? no() : false;
+};
+
+const isMatch = (yes: yesable, no: noable): boolean => {
+    const teams = config.referencesOfType('pa-football-team');
+    const match = config.get('page.footballMatch', {});
+    // the order of this is important as, on occasion,
+    // "minbymin" is tagged with "match reports" but should be considered "minbymin".
+    const pageTypes = [
+        ['minbymin', config.page.isLiveBlog],
+        ['report', config.hasTone('Match reports')],
+        ['preview', config.hasSeries('Match previews')],
+        ['stats', match.id],
+        [null, true], // default
+    ];
+
+    Object.assign(match, {
+        date: config.webPublicationDateAsUrlPart(),
+        teams,
+        isLive: config.page.isLive,
+        pageType: pageTypes.find(type => type[1] === true),
+    });
+
+    if (Array.isArray(match.pageType)) {
+        match.pageType = match.pageType[0];
+    }
+
+    return isit(
+        match.id || (match.pageType[0] && match.teams.length === 2),
```

Comment:
I think if `pageTypes.find(type => type[1] === true)` matches nothing, then `match.pageType` becomes `undefined`, so `match.pageType[0]` will throw an error. But, yeah, the point remains 😄